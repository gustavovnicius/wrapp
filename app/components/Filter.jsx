import React, {
  PropTypes,
} from 'react';
import {
  Radio,
} from 'rebass';
import {
  Flex,
  Box,
} from 'reflexbox';

export default function Filter({ enabled, selected, applyFilter }) {
  return (
    <Flex align="center" justify="center">
      {
        enabled.map(filter => (
          <Box p={2} key={filter.label}>
            <Radio
              circle
              checked={selected.label === filter.label}
              onChange={() => applyFilter(filter)}
              label={filter.label}
              name="filter"
            />
          </Box>
          ))
      }
    </Flex>
  );
}

Filter.propTypes = {
  enabled: PropTypes.array.isRequired,
  selected: PropTypes.object.isRequired,
  applyFilter: PropTypes.func.isRequired,
};
