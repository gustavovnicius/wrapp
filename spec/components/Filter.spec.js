import React from 'react';
import renderer from 'react-test-renderer';
import {
  shallow,
} from 'enzyme';
import {
  Radio,
} from 'rebass';
import Filter from 'components/Filter';

describe('Filter', () => {
  let spy;
  let enabled;
  let selected;

  beforeEach(() => {
    spy = jest.fn();
    enabled = [
      {
        label: 'All',
        filter: jest.fn(),
      },
      {
        label: 'Positive',
        filter: jest.fn(),
      },
    ];
    selected = {
      label: 'All',
      filter: jest.fn(),
    };
  });

  it('match the snapshot', () => {
    const filter = renderer.create(
      <Filter enabled={enabled} selected={selected} applyFilter={spy} />,
    ).toJSON();

    expect(filter).toMatchSnapshot();
  });

  describe('component events', () => {
    let filter;

    beforeEach(() => {
      filter = shallow(
        <Filter enabled={enabled} selected={selected} applyFilter={spy} />,
      );
    });

    it('call applyFilter on a filter selection', () => {
      const expected = {
        label: 'Positive',
        filter: expect.any(Function),
      };

      filter.find(Radio).last().simulate('change');

      expect(spy).toHaveBeenCalledWith(expected);
    });
  });
});
