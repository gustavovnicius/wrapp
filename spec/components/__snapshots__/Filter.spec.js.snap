// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter match the snapshot 1`] = `
<div
  className="Flex"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "boxSizing": "border-box",
      "display": "flex",
      "justifyContent": "center",
      "msAlignItems": "center",
      "msJustifyContent": "center",
    }
  }
>
  <div
    className="Box"
    style={
      Object {
        "boxSizing": "border-box",
        "padding": 16,
      }
    }
  >
    <label
      className="Label"
      style={
        Object {
          "alignItems": "center",
          "boxSizing": "border-box",
          "color": null,
          "cursor": "pointer",
          "display": "flex",
          "flexDirection": null,
          "fontSize": 14,
          "fontWeight": 600,
          "lineHeight": 1,
          "paddingBottom": 8,
          "position": "relative",
        }
      }
    >
      <input
        checked={true}
        name="filter"
        onChange={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "zIndex": -1,
          }
        }
        type="radio"
      />
      <div
        className="Radio_dot"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "currentcolor",
            "borderRadius": 99999,
            "borderStyle": "solid",
            "borderWidth": 5,
            "boxSizing": "border-box",
            "height": 16,
            "marginBottom": null,
            "marginRight": 8,
            "opacity": null,
            "transition": "border .1s ease-out",
            "width": 16,
          }
        }
      />
      All
    </label>
  </div>
  <div
    className="Box"
    style={
      Object {
        "boxSizing": "border-box",
        "padding": 16,
      }
    }
  >
    <label
      className="Label"
      style={
        Object {
          "alignItems": "center",
          "boxSizing": "border-box",
          "color": null,
          "cursor": "pointer",
          "display": "flex",
          "flexDirection": null,
          "fontSize": 14,
          "fontWeight": 600,
          "lineHeight": 1,
          "paddingBottom": 8,
          "position": "relative",
        }
      }
    >
      <input
        checked={false}
        name="filter"
        onChange={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "zIndex": -1,
          }
        }
        type="radio"
      />
      <div
        className="Radio_dot"
        style={
          Object {
            "backgroundColor": "currentcolor",
            "borderColor": null,
            "borderRadius": 99999,
            "borderStyle": null,
            "borderWidth": 5,
            "boxSizing": "border-box",
            "height": 16,
            "marginBottom": null,
            "marginRight": 8,
            "opacity": 0.25,
            "transition": "border .1s ease-out",
            "width": 16,
          }
        }
      />
      Positive
    </label>
  </div>
</div>
`;
